string.c

```diff
@@ -12126,6 +12050,8 @@ rb_enc_interned_str_cstr(const char *ptr, rb_encoding *enc)
  *  - #rstrip, #rstrip!: strip trailing whitespace.
  *  - #strip, #strip!: strip leading and trailing whitespace.
  *
+ *  :include: doc/string/slices.rdoc
```

Comment:
Why this section in a separate file and included? Why can't the contents of slices.rdoc be directly inserted here?

Reply:
The separate file contains non-ascii characters, which are not allowed in C code, even in comments.

Reply:
Rather than having snippets of the documentation in `slides.rdoc` and including it here, maybe we should move this whole thing in its own file. I think that would make it easier to navigate since we don't have to jump between `string.c` and `doc/string/slices.rdoc`.

Reply:
Not sure what you mean by "this whole thing."  We already have 30 inclusions, in behalf of 30 different methods.  (The first one put there by nobu.)

Reply:
I meant "whole thing" as in this whole comment section, starting from the line

> A \String object has an arbitrary sequence of bytes,

and ending with the line

> - #upto: Calls the given block with each string value returned by successive calls to #succ.

Reply:
> I meant "whole thing" as in this whole comment section, starting from the line
> 
> > A \String object has an arbitrary sequence of bytes,
> 
> and ending with the line
> 
> > * #upto: Calls the given block with each string value returned by successive calls to #succ.

I think that would be a first.  A very few C-coded classes have their class doc in a Ruby file;  I don't think any have their class doc in a .rdoc file.

Reply:
What about moving the class doc for String to string.rb?  Regardless of whether string.rdoc or string.rb, can we review/approve/merge this first, then restructure in a new PR?

Reply:
Sure. I'll review this first and then we can do the move in a future PR.

Reply:
> Sure. I'll review this first and then we can do the move in a future PR.

Fwiw, I think my charter is to do as has been done before; therefore I'll be more comfortable moving to string.rb, rather than string.rdoc.